cmd/provisioner/main.go

```diff
@@ -17,103 +17,46 @@
 package main
 
 import (
-	"encoding/json"
 	"fmt"
-	"github.com/Shopify/sarama"
+	"github.com/projectriff/kafka-provisioner/pkg/provisioner/handler"
+	client "github.com/projectriff/kafka-provisioner/pkg/provisioner/kafka"
 	"log"
 	"net/http"
 	"os"
-	"strings"
-)
-
-var (
-	gateway = os.Getenv("GATEWAY")
-	broker  = os.Getenv("BROKER")
 )
 
 func main() {
-
+	gateway := os.Getenv("GATEWAY")
 	if gateway == "" {
 		log.Fatal("Environment variable GATEWAY should contain the host and port of a liiklus gRPC endpoint")
 	}
+	broker := os.Getenv("BROKER")
 	if broker == "" {
 		log.Fatal("Environment variable BROKER should contain the host and port of a Kafka broker")
 	}
-
-	sarama.Logger = log.New(os.Stdout, "[Sarama] ", log.LstdFlags)
-
 	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		if r.Method == http.MethodPut {
-			handlePut(w, r)
-		} else {
+		if r.Method != http.MethodPut {
 			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
 		}
+		handleCreationRequest(broker, gateway, w, r)
```

Comment:
Not always a "Creation"...

Reply:
from the requester point of view, it is, though, isn't it?

Reply:
No, hence the use of PUT. Will rename